### PR TITLE
Table and column names

### PR DESCRIPTION
**Pull request by jheizer**
Allow the defining of table and column names to be added as attributes to the generated code.

### CODE REVIEW

**Comment by pdevito3** [comment]
Ha, great minds 🙂 I actually just finished adding the capability to add table names and schemas for the next release (check out the [v6.x branch](https://github.com/pdevito3/craftsman/tree/v0.6.x). If you want you can go ahead and try to merge it in, otherwise, I'll leave this and try to merge before the next release as there's definitely some overlap. 

Regardless, appreciate the contributions!! Let me know if you want to work on any more features so we can sync up on things.

File: Craftsman.Tests/FileTextTests/EntityFileTextTests.cs [hunk 1]
```diff
@@ -83,5 +83,82 @@ public class Product
...
+        [Fact]
+        public void GetEntityFileText_passed_custom_property_column_name_entity_creates_expected_text()
+        {
+            var classNamespace = "Domain.Entities";
+            var entity = CannedGenerator.FakeBasicProduct();
+            entity.Properties.Add(new EntityProperty { Name = "ProductType", Type = "int", CanFilter = true, CanSort = true, ColumnName = "Product_Type" });
+
+            var fileText = EntityBuilder.GetEntityFileText(classNamespace, entity);
+
+            var expectedText = @$"namespace Domain.Entities
```
Comment (pdevito3) [inline Craftsman.Tests/FileTextTests/EntityFileTextTests.cs:135]: Is this an accident?

File: Craftsman.Tests/FileTextTests/EntityFileTextTests.cs [hunk 2]
```diff
@@ -83,5 +83,82 @@ public class Product
...
+        [Sieve(CanFilter = true, CanSort = false)]
+        public string Name {{ get; set; }}
+
+        [Sieve(CanFilter = true, CanSort = true)]
+        [Column(""Product_Type"")]
+        public int ProductType {{ get; set; }}
+
+        // add-on property marker - Do Not Delete This Comment
+    }}
+}}";
```
Comment (pdevito3) [inline Craftsman.Tests/FileTextTests/EntityFileTextTests.cs:159]: This also looks like an accident?

File: Craftsman/Builders/EntityBuilder.cs [hunk 3]
```diff
@@ -52,14 +52,15 @@ public static string GetEntityFileText(string classNamespace, Entity entity)
             var usingSieve = entity.Properties.Where(e => e.CanFilter == true || e.CanSort == true).ToList().Count > 0 ? @$"{Environment.NewLine}    using Sieve.Attributes;" : "";
             var inheritanceString = entity.Auditable ? $" : AuditableEntity" : "";
             var auditableUsing = entity.Auditable ? @$"{Environment.NewLine}    using Domain.Common;" : "";
+            var tableName = !string.IsNullOrEmpty(entity.TableName) ? @$"[Table(""{entity.TableName}"")]{Environment.NewLine}    " : "";
 
             return @$"namespace {classNamespace}
 {{
```
Comment (pdevito3) [inline Craftsman/Builders/EntityBuilder.cs:58]: Also a typo?

File: Craftsman/Models/Entity.cs [hunk 4]
```diff
@@ -61,5 +61,10 @@ public EntityProperty PrimaryKeyProperty
         /// When set to true, will have the entity inherit from 'AuditableEntity'. Defaults to false.
         /// </summary>
         public bool Auditable { get; set; } = false;
+
+        /// <summary>
```
Comment (pdevito3) [inline Craftsman/Models/Entity.cs:65]: Could you remove table name? I've already added that separately.

File: Craftsman/Builders/EntityBuilder.cs [hunk 5]
```diff
@@ -52,14 +52,15 @@ public static string GetEntityFileText(string classNamespace, Entity entity)
...
     using System;
     using System.ComponentModel.DataAnnotations;
     using System.ComponentModel.DataAnnotations.Schema;{usingSieve}{auditableUsing}
 
-    public class {entity.Name}{inheritanceString}
+    {tableName}public class {entity.Name}{inheritanceString}
```
Comment (pdevito3) [inline Craftsman/Builders/EntityBuilder.cs:63]: Could you remove table name? I've already added that separately.

File: Craftsman/Builders/EntityBuilder.cs [hunk 6]
```diff
@@ -100,6 +101,8 @@ private static string AttributeBuilder(EntityProperty entityProperty)
                 attributeString += @$"        [ForeignKey(""{entityProperty.ForeignKeyPropName}"")]{Environment.NewLine}";
             if (entityProperty.CanFilter || entityProperty.CanSort)
                 attributeString += @$"        [Sieve(CanFilter = {entityProperty.CanFilter.ToString().ToLower()}, CanSort = {entityProperty.CanSort.ToString().ToLower()})]{Environment.NewLine}";
+            if (!string.IsNullOrEmpty(entityProperty.ColumnName))
```
Comment (pdevito3) [inline Craftsman/Builders/EntityBuilder.cs:104]: +1

File: Craftsman/Models/EntityProperty.cs [hunk 7]
```diff
@@ -92,5 +92,11 @@ public bool IsForeignKey
         /// Default value for this property
         /// </summary>
         public string DefaultValue { get; set; }
+
```
Comment (pdevito3) [inline Craftsman/Models/EntityProperty.cs:95]: +1

**Review by pdevito3 — CHANGES_REQUESTED** [review]
Hey, looks like there might have been a couple of characters accidentally deleted. Otherwise, could you just removed the tablename feature as I already added that in.

**Comment by jheizer** [comment]
Hmm, I'm confused.  It might be github not liking the embedded strings with {{.  I see as red just reading the source on here.  Ex: https://github.com/pdevito3/craftsman/blob/master/Craftsman/Builders/EntityBuilder.cs  I don't see that I have changes on the given lines.

**Comment by pdevito3** [comment]
Yeah, I just looked at the PR on git fork to merge it in and those weren't showing up. Was about to comment to let you know. 

Don't worry about this, I just conflict resolved merge into v0.6 to tweak the tests a smidge to work for some controller updates i made. You should get the credit for the commits, so let me know if there is an issue

**Review by pdevito3 — APPROVED** [review]
looks good! I'll be sure to merge these in so you get credit. could you just change the merge branch from `pdevito3:master` to `pdevito3:v0.6.x`?

**Comment by jheizer** [comment]
Pushed changes in there, hopefully didn't mess things up.  It showing those characters in red sure is annoying

**Pull request merged**